### DISCUSSION TITLE
Image doesn't fit in partition length

### DISCUSSION THREAD

**Discussion by icho40** [pschatzmann/arduino-audio-tools#1815] [Q&A] [answered]
Hello,

I would like to build an MP3 player with ESP32.
But when I flash the corresponding image,
I get this error message:

```
(SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0030,len:4604
ho 0 tail 12 room 4
load:0x40078000,len:15488
load:0x40080400,len:4
load:0x40080404,len:3180
entry 0x400805b8
E (474) esp_image: Image length 1312592 doesn't fit in partition length 1310720
E (474) boot: OTA app partition slot 0 is not bootable
E (474) esp_image: image at 0x150000 has invalid magic byte (nothing flashed here?)
E (482) boot: OTA app partition slot 1 is not bootable
E (487) boot: No bootable app partitions in the partition table
ets Jul 29 2019 12:21:46
```

With other images it works:
base-adc-measure		no error
base-file_raw-serial	no error
send-8bit-receive		no error
player-sd-i2s			no error

Arduino IDE 2.3.3 (AppImage)
Board: DOIT ESP32 DEVKIT V1

Does anyone know what to do?

(translated from German with deepl.com)

**Comment by pschatzmann** [comment, marked as answer]
Did you eventually select a partition size where it would not fit ?

**Reply by icho40** [reply]
not that I know of, at least not intentionally.
I only ever took the unchanged inos from the examples.

**Comment by pschatzmann** [comment]
So I guess that your problem: The default setting is quite bad!

**Reply by icho40** [reply]
Surprisingly, it worked with the others

**Reply by icho40** [reply]
Can you please tell me where I need to change this?
I can't find anything in the ino file player-sdfat-analog.ino.

**Reply by icho40** [reply]
Thanks to your hint I found my mistake.
The setting in Arduino was
"Erase all flash... disabled
after I changed it to enabled,
it worked. 


Thanks for your help and many great examples.

**Reply by pschatzmann** [reply]
The partition scheme can be set in the Tools menu. 
I recommend to use Huge App for sketches that use a lot of PROGMEM...
